### DISCUSSION TITLE
Should registration information of Kyuubi in ZK be cleaned up？

### DISCUSSION THREAD

**Discussion by merrily01** [apache/kyuubi#4578] [General]
Since I started testing the Kyuubi + Zookeeper method, it seems that the registration information of Kyuubi in ZK has never been cleaned up, and some of the services or engines have been disabled for a long time.
e.g.
![image](https://user-images.githubusercontent.com/15688792/226792586-1fa52afb-c7ae-48d6-abc3-d88d6c09301e.png)

![image](https://user-images.githubusercontent.com/15688792/226793007-0c270cda-8c01-4563-a33c-51b53d3aff66.png)

Is this consistent with the design, or should there be any cleanup mechanisms necessary?

**Comment by yaooqinn** [comment]
I can not recall my original design. :)

**Reply by yaooqinn** [reply]
It should be deleted during the engine termination.

**Reply by merrily01** [reply]
Thx，I think it should be deleted. 

But in some scenarios, it may not seem as expected. 

However, I haven't been able to determine which scenarios may have this issue. 😓

I think I can keep testing and following it.

**Reply by yaooqinn** [reply]
thanks for taking care of it
